### DISCUSSION TITLE
Questions: Is it possible to run the reinforcement learning experiment without printing out the record?

### DISCUSSION THREAD

**Discussion by Ttopiac** [rlworkgroup/garage#2268] [General]
I am training the robotic model to solve a task with TRPO by using the following two libraries.

_**From garage.trainer import Trainer
from garage.torch.algos import TRPO**_

During the experiment, programs will print the logs. However, I want it not to do so in order to save more experiment time. Is it possible to only run the experiment without printing out the experiment logs? 

I am talking about the logs like the following. Thank you for reading my questions.

_Evaluation/AverageDiscountedReturn    -3499.43
Evaluation/AverageReturn             -19772.9
Evaluation/Iteration                    901
Evaluation/MaxReturn                 -19419.8
Evaluation/MinReturn                 -20069.7
Evaluation/NumEpisodes                    4
Evaluation/StdReturn                    241.69
Evaluation/TerminationRate                1
GaussianMLPPolicy/Entropy                34.2077
GaussianMLPPolicy/KL                      0.00994284
GaussianMLPPolicy/KLBefore                0
GaussianMLPPolicy/LossAfter              22.0052
GaussianMLPPolicy/LossBefore             35.2744
GaussianMLPPolicy/dLoss                  13.2692
GaussianMLPValueFunction/LossAfter        8.19468
GaussianMLPValueFunction/LossBefore       8.19468
GaussianMLPValueFunction/dLoss            9.53674e-07
TotalEnvSteps                             3.608e+06_

**Comment by krzentner** [comment]
It's very unlikely that logging has any measurable affect on performance, but if you don't want to save logs, you [can set `trainer.enable_logging` to false](https://github.com/rlworkgroup/garage/blob/master/src/garage/trainer.py#L131)

**Reply by krzentner** [reply]
It's also worth considering that whatever application you have in mind, you most likely need to check the log anyways to make sure that the training didn't produce a broken policy. The only case where you don't is if you're training one of the easy gym or atari environments with known hyperparameters. Unfortunately deep reinforcement learning as a field is not yet at a point where it reliably produces working policies to new environments without human supervision.
